src/ImportDefinitionsBundle/Provider/ExportProviderInterface.php

```diff
@@ -23,5 +23,14 @@
      * @param $params
      * @return array
      */
-    public function exportData(array $data, $configuration, $definition, $params);
+    public function addExportData(array $data, $configuration, $definition, $params);
+
+    /**
+     * @param array $configuration
+     * @param $definition
+     * @param $params
+     * @return array
+     */
+    public function exportData($configuration, $definition, $params);
+
```

Comment:
ExportInterface now has two methods instead of only one. With addExportData we add the export data to the provider one object at a time. exportData is called at the very end to write all the added data to "file". For a "streaming" export provider the data could be appended to "file" already in addExportData.